platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/roster/RosterValidatorTests.java

```diff
@@ -0,0 +1,305 @@
+                                RosterEntry.newBuilder()
+                                        .nodeId(2)
+                                        .weight(1)
+                                        .gossipCaCertificate(Bytes.wrap("test"))
+                                        .tssEncryptionKey(Bytes.wrap("test"))
+                                        .gossipEndpoint(ServiceEndpoint.newBuilder()
+                                                .domainName("domain.com")
+                                                .port(666)
+                                                .build())
+                                        .build())
+                        .build()));
+    }
+
+    @Test
+    void invalidGossipCertTest() {
+        assertThrows(
+                InvalidRosterException.class,
+                () -> RosterValidator.validate(Roster.newBuilder()
+                        .rosters(
+                                RosterEntry.newBuilder()
+                                        .nodeId(1)
+                                        .weight(1)
+                                        .gossipCaCertificate(Bytes.wrap("test"))
+                                        .tssEncryptionKey(Bytes.wrap("test"))
+                                        .gossipEndpoint(ServiceEndpoint.newBuilder()
+                                                .domainName("domain.com")
+                                                .port(666)
+                                                .build())
+                                        .build(),
+                                RosterEntry.newBuilder()
+                                        .nodeId(2)
+                                        .weight(1)
+                                        .gossipCaCertificate(Bytes.EMPTY)
+                                        .tssEncryptionKey(Bytes.wrap("test"))
+                                        .gossipEndpoint(ServiceEndpoint.newBuilder()
+                                                .domainName("domain.com")
+                                                .port(666)
+                                                .build())
+                                        .build(),
+                                RosterEntry.newBuilder()
+                                        .nodeId(3)
+                                        .weight(1)
+                                        .gossipCaCertificate(Bytes.wrap("test"))
+                                        .tssEncryptionKey(Bytes.wrap("test"))
+                                        .gossipEndpoint(ServiceEndpoint.newBuilder()
+                                                .domainName("domain.com")
+                                                .port(666)
+                                                .build())
+                                        .build())
+                        .build()));
+    }
+
+    @Test
+    void emptyGossipEndpointsTest() {
+        assertThrows(
```

Comment:
tests look good. One thing i'd suggest is to assert specific exception message you want instead of blanket assertthrows.

Reply:
Cool, added message checks.